### PR TITLE
Deprecate AWS_VPC_K8S_CNI_CONFIGURE_RPFILTER and remove no-op setter

### PR DESCRIPTION
**Pull request by jdn5126**
**What type of PR is this?**
Cleanup

**Which issue does this PR fix**:
None

**What does this PR do / Why do we need it**:
This PR removes an unnecessary rp_filter setter in `pkg/networkutils/network.go` and deprecates the `AWS_VPC_K8S_CNI_CONFIGURE_RPFILTER` environment variable. During IPAMD node init, we were setting `rp_filter` on primary interface to "loose" mode in certain scenarios. But `rp_filter` is already set to "loose" mode on primary interface unconditionally by init container.

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:
N/A

**Testing done on this change**:
Manually verified that canary tests pass and that `rp_filter` is set before this code even executes.

**Automation added to e2e**:
N/A

**Will this PR introduce any new dependencies?**:
No

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
Will not affect upgrade or downgrades. A running cluster has been tested.

**Does this change require updates to the CNI daemonset config files to work?**:
No

**Does this PR introduce any user-facing change?**:
No

```release-note
Deprecating AWS_VPC_K8S_CNI_CONFIGURE_RPFILTER environment variable.
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jdn5126** [comment]
The unconditional setting of `rp_filter` to "loose" mode in init container happens here btw: https://github.com/aws/amazon-vpc-cni-k8s/blob/master/cmd/aws-vpc-cni-init/main.go#L73

**Comment by jayanthvn** [comment]
Toggling `AWS_VPC_K8S_CNI_CONFIGURE_RPFILTER` is tested?

This is called out in docs -

```
Specifies whether ipamd should configure rp filter for primary interface. Setting this to false will require rp filter to be configured through init container
```

**Comment by jdn5126** [comment]
> 

Yeah, so that environment variable becomes a no-op with this change. I was going to remove it, but wanted to wait first. I should probably just remove it as part of this change.

Previously, VPC CNI could be run without the init container, but today, we require the init container to do required setup. So since the init container is required and unconditionally sets `rp_filter` to `2`, this logic is not needed.

File: pkg/networkutils/network.go [hunk 1]
```diff
@@ -273,39 +270,7 @@ func (n *linuxNetwork) SetupHostNetwork(vpcv4CIDRs []string, primaryMAC string,
...
-		//   the RPF check, it looks up the route using the pod IP as the source.
-		// - Thus, it finds the source-based route that leaves via the secondary ENI.
-		// - In "strict" mode, the RPF check fails because the return path uses a different interface to the incoming
-		//   packet. In "loose" mode, the check passes because some route was found.
-		//
-		// The same applies when performing SNAT on Pod traffic.
-		primaryIntfRPFilter := "net/ipv4/conf/" + primaryIntf + "/rp_filter"
-		const rpFilterLoose = "2"
-
-		if n.shouldConfigureRpFilter {
```
Comment (kishorj) [inline pkg/networkutils/network.go:298]: Also cleanup shoudConfigureRpFilter attribute, the underlying env variable and documentation.

File: README.md [hunk 2]
```diff
@@ -390,13 +390,15 @@ string, tag addition will be ignored.
 
 ---
 
-#### `AWS_VPC_K8S_CNI_CONFIGURE_RPFILTER`
```
Comment (jaydeokar) [inline README.md:393]: Would this be 1.12.1 ?

Comment (jdn5126) [reply README.md:393]: good call, fixed

**Pull request merged**